### DISCUSSION TITLE
WebView2 Exception "The function should be called before any other API is called in 'CoreWebView2Environment' class"

### DISCUSSION THREAD

**Discussion by kentlhastings** [MicrosoftEdge/WebView2Feedback#4009] [Q&A]
We are using WebView2 to display html content within AutoCAD 2022. Users run an AutoCAD command to display the WebView2 UI within a WPF window. For some of our users, after the WebView2 UI is run once and then closed, upon running the command to launch again, they receive the following exception message: "The function should be called before any other API is called in 'CoreWebView2Environment' class".

Here is the code we are using to initialize the UI:
![image](https://user-images.githubusercontent.com/58216656/210820461-b52eb96a-2379-48b2-a726-2d6ad5dcec90.png)

Here is the code we are using to shutdown the UI:
![image](https://user-images.githubusercontent.com/58216656/210820658-651cc78b-147b-4353-afe0-556ac12cf70b.png)

Any insight you could provide for how to avoid this problem would be greatly appreciated

Thank you,

**Comment by victorthoang** [comment]
Hello @kentlhastings,

Thanks for your inquiry! I've assigned this to a dev that might best answer your question.

**Comment by novac42** [comment]
Thanks for providing the code snippet and we will take a look into it. Currently our team is a bit short on bandwidth and will try to give an update this month.

**Comment by LiangTheDev** [comment]
The message doesn't appear to be coming from WebView2 code, and also not sure about the Browser class. @kentlhastings if this is still an issue, could you please provide the full stack of the exception?
